sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/batch/BulkExecutor.java

```diff
@@ -134,20 +143,42 @@ public BulkExecutor(CosmosAsyncContainer container,
             .doOnComplete(() -> {
                 mainSourceCompleted.set(true);
 
-                if (totalCount.get() == 0) {
+                long totalCountSnapshot = totalCount.get();
+                logger.debug("Main source completed - # left items " + totalCountSnapshot);
```

Comment:
please try to use logger placeholder to avoid unnecessary string concat

```suggestion
                logger.debug("Main source completed - # left items {}", totalCountSnapshot);
```